daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -151,27 +151,41 @@ object TransactionCoder {
         }
 
       case e: NodeExercises[Nid, Cid, Val] =>
-        encodeVal(e.chosenValue).map {
-          case (vversion, arg) =>
-            val exBuilder =
-              TransactionOuterClass.NodeExercise
-                .newBuilder()
-                .setChoice(e.choiceId)
-                .setTemplateId(ValueCoder.encodeIdentifier(e.templateId, Some(vversion))._2)
-                .setChosenValue(arg)
-                .setConsuming(e.consuming)
-                .setContractIdOrStruct(encodeCid, transactionVersion, e.targetCoid)(
-                  _.setContractId(_),
-                  _.setContractIdStruct(_))
-                .addAllActors(e.actingParties.map(_.underlyingString).asJava)
-                .addAllChildren(e.children.map(encodeNid).toList.asJava)
-                .addAllControllers(e.controllers.map(_.underlyingString).asJava)
-                .addAllSignatories(e.signatories.map(_.underlyingString).asJava)
-                .addAllStakeholders(e.stakeholders.map(_.underlyingString).asJava)
+        for {
+          argValue <- encodeVal(e.chosenValue)
+          retValue <- e.exerciseResult
```

Comment:
```suggestion
          import scalaz.std.option._ // at top of file
          retValue <- e.exerciseResult traverseU encodeVal
          // and delete the map/getOrElse
```

Reply:
More of an organizational problem, but: here we unconditionally fail if encoding `exerciseResult` fails. But as described in another comment, we don't care whether encoding `exerciseResult` succeeds or not if  `transactionVersion precedes ...`. So this probably needs to move closer to the point of use.